lib/AST/Decl.cpp

```diff
@@ -2450,6 +2450,10 @@ DestructorDecl *ClassDecl::getDestructor() {
 }
 
 bool ClassDecl::inheritsSuperclassInitializers(LazyResolver *resolver) {
+  // Get a resolver from the ASTContext if we don't have one already.
+  if (resolver == nullptr)
+    resolver = getASTContext().getLazyResolver();
```

Comment:
Can you describe a scenario where code that passed in null becomes incorrect?

Removing the parameter altogether and always getting the resolver from the context as we've done elsewhere already would be a nice simplification, but I wanted to do a simple fix that's suitable for 3.1 first.